### PR TITLE
Added client call to empty handler when chat is complete

### PR DESCRIPTION
**Pull request by replaceafill**
This PR adds an Xblock.handler and calls it from the JS client code when the chat is complete.

**JIRA ticket**: [OC-2534](https://tasks.opencraft.com/browse/OC-2534)

**Testing instructions**:

1. Add a new Chat block in Studio with the default sample steps.
2. In the LMS navigate to the XBlock and answer "No, not right now" which will trigger the final step.
3. Before clicking the "Bye!" response button open the Network tab in the Chrome Developer Tools.
4. After clicking the "Bye!" response button the "chat_complete" handler call should be displayed in the tools.

**Note**:

This xblock stores state into `localStorage` in addition to the server, so when you're testing these changes, you will find that the "Delete learner's state" button from "Staff Debug" tools doesn't reset the problem. To completely reset the problem, you should invoke `localStorage.clear()` in the JS console in addition to invoking "Delete learner's state" staff debug tool.

**Reviewers**
- [x] @bdero

### CODE REVIEW

File: chat/chat.py
Comment (bradenmacdonald) [inline chat/chat.py]: Maybe mention that "Does nothing at the moment; this HTTP request is listened for by some mobile apps to trigger events after the chat is completed."

File: chat/public/js/src/chat.js
Comment (mtyaka) [inline chat/public/js/src/chat.js]: In javascript, empty array is not falsey, so I think you need to check whether its length is zero.

File: tests/integration/test_chat.py
Comment (mtyaka) [inline tests/integration/test_chat.py]: It would be good to test [this case](https://github.com/replaceafill/xblock-chat/blob/318c3e410d96a3fd7fdfc56b86514606d8bb15d6/tests/integration/test_chat.py#L226), too.

File: chat/chat.py
Comment (replaceafill) [reply chat/chat.py]: Thanks. Done.

File: chat/public/js/src/chat.js
Comment (replaceafill) [reply chat/public/js/src/chat.js]: Ah, thanks!

File: tests/integration/test_chat.py
Comment (replaceafill) [reply tests/integration/test_chat.py]: Thanks. Done.

File: chat/public/js/src/chat.js
Comment (bdero) [inline chat/public/js/src/chat.js]: Microscopic grammar nit: "exists" should be "exist" here. ;)

**Review by bdero — APPROVED** [review]
:+1:

- [x] I tested this: I installed this version of the chat xblock and witnessed that the `chat_complete` handler is called when the chat is complete via the Chromium inspector. This happens in the case where there are no responses available, or if the step that the user is directed to doesn't exist (a completion step).
- [x] I read through the code
- [x] ~~I checked for accessibility issues~~ N/A. No accessibility impact.
- [x] Includes documentation - the inline comments are very detailed

**Comment by bradenmacdonald** [comment]
@replaceafill or one of the reviewers, can you please place an example of the full URL for the `chat_complete` event? something like http://foo/bar/xblock_handler/chat_complete I assume?

**Comment by bdero** [comment]
@bradenmacdonald Here's an example URL from my devstack:
```
http://localhost:8000/courses/course-v1:asdf+asdf+asdf/xblock/block-v1:asdf+asdf+asdf+type@chat+block@809ec1f3ba364473b24402fc0491f17f/handler/chat_complete
```

**Comment by bradenmacdonald** [comment]
@bdero perfect, thanks!

File: chat/public/js/src/chat.js
Comment (replaceafill) [reply chat/public/js/src/chat.js]: Thanks! Addressed.

**Comment by replaceafill** [comment]
@bdero a couple of questions:

1. Should I rebase all these commits into a single commit before merging?
2. I don't have edit access to the open-craft/xblock-chat repo so I don't see the Merge button. Can you either allow me to or merge this please?

Thanks.

**Comment by bdero** [comment]
@replaceafill Ah, I can hit the merge button after you squash the commits. Thanks!

**Comment by replaceafill** [comment]
@bdero Squashed. Thanks a lot!

**Comment by bdero** [comment]
@replaceafill Merged, thanks!

**Pull request merged**
